molecule/testinfra/common/test_grsecurity.py

```diff
@@ -99,39 +103,46 @@ def test_grsecurity_sysctl_options(host, sysctl_opt):
         assert host.sysctl(sysctl_opt[0]) == sysctl_opt[1]
 
 
-@pytest.mark.skip_in_prod
-@pytest.mark.parametrize('paxtest_check', [
-  "Executable anonymous mapping",
-  "Executable bss",
-  "Executable data",
-  "Executable heap",
-  "Executable stack",
-  "Executable shared library bss",
-  "Executable shared library data",
-  "Executable anonymous mapping (mprotect)",
-  "Executable bss (mprotect)",
-  "Executable data (mprotect)",
-  "Executable heap (mprotect)",
-  "Executable stack (mprotect)",
-  "Executable shared library bss (mprotect)",
-  "Executable shared library data (mprotect)",
-  "Writable text segments",
-  "Return to function (memcpy)",
-  "Return to function (memcpy, PIE)",
-])
-def test_grsecurity_paxtest(host, paxtest_check):
+def test_grsecurity_paxtest(host):
     """
-    Check that paxtest does not report anything vulnerable
-    Requires the package paxtest to be installed.
-    The paxtest package is currently being installed in the app-test role.
+    Check that paxtest reports the expected mitigations. These are
+    "Killed" for most of the checks, with the notable exception of the
+    memcpy ones. Only newer versions of paxtest will fail the latter,
+    regardless of kernel.
     """
-    if host.exists("/usr/bin/paxtest"):
+    if not host.exists("/usr/bin/paxtest"):
+        warnings.warn("Installing paxtest to run kernel tests")
+        with host.sudo():
+            host.run("apt-get install -y paxtest")
```

Comment:
Assumes that paxtest is available via apt - which may not be the case until `apt-get update` has run at least once on the host. Running `apt get update && apt-get install -y paxtest` would work.

Reply:
Added!